runtime/src/main/java/org/corfudb/runtime/collections/PersistedStreamingMap.java

```diff
@@ -318,9 +274,13 @@ public void clear() {
         // If it is nullptr, the iterator will read from an implicit snapshot as of the time the
         // iterator is created. The implicit snapshot is preserved by pinning resource.
         readOptions.setSnapshot(null);
-        final RocksIterator rocksIterator = rocksDb.newIterator(readOptions);
-        rocksIterator.seekToFirst();
-        return Streams.stream(new RocksDbIterator(rocksIterator));
+        final RocksDbEntryIterator entryIterator =
+                new RocksDbEntryIterator(rocksDb.newIterator(readOptions), serializer);
+        // It's necessary to use the RocksDbEntryIterator as try-with resource to ensure
+        // the iterator is close, otherwise underlying RocksDB resources can leak
+        Stream<Entry<K, V>> resStream = Streams.stream(entryIterator);
+        resStream.onClose(entryIterator::close);
```

Comment:
According to this: https://github.com/facebook/rocksdb/blob/master/java/src/main/java/org/rocksdb/ReadOptions.java#L11
I think we need to call `readOptions.close()` before returning to prevent memory leak. (dispose is deprecated, so using close())

Reply:
Didn't know about this. Thanks!

It will get released through finalized, but we shouldn't relay on that.

Reply:
Done.